utilities/partitioning/partition_validator.py

```diff
@@ -0,0 +1,167 @@
+                inter = cachment_set1.intersection(catchment_set2)
+                if len(inter) > 0:
+                    cat_id_conflicts = cat_id_conflicts + 1
+                    print("partition " + str(key1) + " and partition " + str(key2) + " both contain" + str(inter))
+
+    return cat_id_conflicts
+
+
+def test_nexus_ids(data: dict):
+    """This function check the nexus ids in the recorded partitions to determine the set of remote nexi
+
+        @param data A dictionary containing a list of partition records stored with the key 'partitions'
+
+        """
+    nex_ids = {}
+
+    # create a set of each partitions catchment_ids
+    for partition in data['partitions']:
+        nex_ids[partition["id"]] = set(partition["nex-ids"])
+
+    remote_nexus_ids = set()
+
+    # find the remote nexi by finding all nexus ids that are more than one partition
+    for key1, cachment_set1 in nex_ids.items():
+        for key2, catchment_set2 in nex_ids.items():
+            if key1 != key2:
+                inter = cachment_set1.intersection(catchment_set2)
+                if len(inter) > 0:
+                    remote_nexus_ids = remote_nexus_ids.union(inter);
+
+    return remote_nexus_ids
+
+
+def test_remote_connections(data: dict):
+    """This function checks to ensure that there exist a minal set of reciving remote connection sections, such
+       that MPI send and recives will pair properly
+
+        @param data A dictionary containing a list of partition records stored with the key 'partitions'
+
+        TODO update logic when partition files encode full communication path in recmote sections
+
+        """
+    partitions = data['partitions']
+
+    mpi_rank = -1;
+    err_count = 0
+    for partition in partitions:
+        mpi_rank = mpi_rank + 1
+        for rc in partition['remote-connections']:
+
+            # get references about the current remote connection
+            target_rank = rc["mpi-rank"]
+            nex_id = rc["nex-id"]
+            cat_id = rc["cat-id"]
+            cat_dir = rc["cat-direction"]
+
+            # currently only check send from upstream to downstream
+            if cat_dir == "orig_cat-to-nex":
+                continue
+
+            try:
+                # get a reference to the target partition based on the specified mpi rank
+                target_part = next(x for x in partitions if x["id"] == target_rank)
+
+                # get the list of remote connections
+                target_connections = target_part['remote-connections']
+
+                # find the matching remote connection
+                matches = [x for x in target_connections
+                           if x["nex-id"] == nex_id and
+                           x["mpi-rank"] == mpi_rank and
+                           cat_dir != x["cat-direction"]]
+                if len(matches) > 0:
+                    # any number of matching records will ensure the needed receive is generated
+                    pass
```

Comment:
This syntax kinda seems odd.  Did you have a specific reason to explicitly `pass` instead of just inverting the test condition?

Reply:
In this case the test is looking for a valid result. However the correct action for a valid result is to do nothing